### PR TITLE
Configure creation destroy timeouts

### PR DESCRIPTION
**Pull request by pablogs9**
This PR enables the possibility of configuring creation, destruction, pub, and req timeouts.

TODO:
- [x] Add tests for req/res

### CODE REVIEW

File: rmw_microxrcedds_c/test/test_client.cpp [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    client = rmw_create_client(
+      this->node,
+      &dummy_type_supports.back().type_support,
+      service_name,
+      &dummy_qos_policies);
+
+    ASSERT_EQ((void *)client, (void *)NULL);
+    ASSERT_EQ(CheckErrorState(), true);
+
+    // Relese one
```
Comment (jamoralp) [inline rmw_microxrcedds_c/test/test_client.cpp:137]:
```suggestion
    // Release one
```

File: rmw_microxrcedds_c/CMakeLists.txt [hunk 2]
```diff
@@ -56,6 +56,8 @@ set(RMW_UXRCE_TOPIC_NAME_MAX_LENGTH "100" CACHE STRING "This value sets the maxi
 set(RMW_UXRCE_TYPE_NAME_MAX_LENGTH "128" CACHE STRING "This value sets the maximum number of characters for a type name.")
 set(RMW_UXRCE_XML_BUFFER_LENGTH "600" CACHE STRING "This value sets the maximum number of characters for a XML buffer.")
 set(RMW_UXRCE_REF_BUFFER_LENGTH "100" CACHE STRING "This value sets the maximum number of characters for a reference buffer.")
+set(RMW_UXRCE_ENTITY_CREATION_DESTROY_TIMEOUT "1000" CACHE STRING "This value sets the maximum time to wait for an XRCE entity creation and destroy.")
+set(RMW_UXRCE_PUBLISH_RELIABLE_TIMEOUT "1000" CACHE STRING "This value sets the maximum time to wait for a publication in a reliable mode.")
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply rmw_microxrcedds_c/CMakeLists.txt]: Done

File: rmw_microxrcedds_c/src/rmw_client.c [hunk 3]
```diff
@@ -206,11 +202,7 @@ rmw_destroy_client(
...
       custom_client->client_id);
 
-    uint16_t requests[] = {delete_client};
-    uint8_t status[sizeof(requests) / 2];
-    if (!uxr_run_session_until_all_status(
-        &custom_node->context->session, 1000, requests, status,
-        sizeof(status)))
+    if (!run_xrce_session(custom_node->context, delete_client))
     {
       RMW_SET_ERROR_MSG("unable to remove client from the server");
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply rmw_microxrcedds_c/src/rmw_client.c]: Done

File: rmw_microxrcedds_c/CMakeLists.txt [hunk 4]
```diff
@@ -56,6 +56,8 @@ set(RMW_UXRCE_TOPIC_NAME_MAX_LENGTH "100" CACHE STRING "This value sets the maxi
 set(RMW_UXRCE_TYPE_NAME_MAX_LENGTH "128" CACHE STRING "This value sets the maximum number of characters for a type name.")
 set(RMW_UXRCE_XML_BUFFER_LENGTH "600" CACHE STRING "This value sets the maximum number of characters for a XML buffer.")
 set(RMW_UXRCE_REF_BUFFER_LENGTH "100" CACHE STRING "This value sets the maximum number of characters for a reference buffer.")
+set(RMW_UXRCE_ENTITY_CREATION_DESTROY_TIMEOUT "1000" CACHE STRING "This value sets the maximum time to wait for an XRCE entity creation and destroy.")
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply rmw_microxrcedds_c/CMakeLists.txt]: Done

File: rmw_microxrcedds_c/test/test_client.cpp (hunk 1 above)
Comment (pablogs9) [reply rmw_microxrcedds_c/test/test_client.cpp]: Done

**Review by jamoralp — APPROVED** [review]

**Pull request merged**